src/backend/cdb/motion/cdbmotion.c

```diff
@@ -1274,3 +1279,115 @@ UpdateSentRecordCache(int32 *sent_record_typmod)
 		*sent_record_typmod = NextRecordTypmod;
 	}
 }
+
+static int
+GetICTypeByName(const char *type_name)
+{
+	for (int i = 0; i < CurrentIPCLayerImplNum; ++i)
+	{
+		if (!pg_strcasecmp(IPCLayerImpls[i]->type_name, type_name))
+			return IPCLayerImpls[i]->ic_type;
+	}
+
+	return -1;
+}
+
+/*
+ * try best to find the most appropriate IPC layer implement.
+ */
+static int
+TryFindICType()
+{
+	/* IPCLayerImpls[0]->ic_type is valid at least */
+	Assert(CurrentIPCLayerImplNum);
+
+	int candidate_types_order[] =
+		{Gp_interconnect_type, INTERCONNECT_TYPE_UDPIFC, IPCLayerImpls[0]->ic_type};
+
+	for (int i = 0; i < (sizeof(candidate_types_order) / sizeof(int)); ++i)
```

Comment:
src/include/c.h:749:#define lengthof(array) (sizeof (array) / sizeof ((array)[0]))

Reply:
fix it with [commit](https://github.com/cloudberrydb/cloudberrydb/pull/646/commits/ea9f6eb1e2ef221af2b48112393eda930a23ff34)